From 438a16b089f152eedb5a4b96f6b98c82635708f1 Mon Sep 17 00:00:00 2001
From: Matt DeVillier <matt.devillier@puri.sm>
Date: Thu, 17 Jun 2021 18:58:45 -0500
Subject: [PATCH 4/4] mb/purism/librem_mini/hda_verb.c: Fix subsystem, jack
 detect

- set subsystem/subvendor ID to Realtek default, as the one dumped
  from the vendor UEFI firmware provides no advantages
- Add a codec reset before setting the subvendor ID using the Azalia macro
  for consistency with all other Realtek HDA codecs
- disable jack detect for the external mic on the 3.5mm jack, since it's not
  currently working, so that the external microphone can be manually selected

Change-Id: Ib0f99e5088973a721c0a295899012c9aea5009cf
Signed-off-by: Matt DeVillier <matt.devillier@puri.sm>
---
 .../librem_cnl/variants/librem_mini/hda_verb.c       | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/src/mainboard/purism/librem_cnl/variants/librem_mini/hda_verb.c b/src/mainboard/purism/librem_cnl/variants/librem_mini/hda_verb.c
index 93dd2702bf..5a9aead2e7 100644
--- a/src/mainboard/purism/librem_cnl/variants/librem_mini/hda_verb.c
+++ b/src/mainboard/purism/librem_cnl/variants/librem_mini/hda_verb.c
@@ -3,16 +3,18 @@
 #include <device/azalia_device.h>
 
 const u32 cim_verb_data[] = {
-	0x10ec0269,	/* Codec Vendor/Device ID: Realtek ALC293 */
-	0x10ec0000,	/* Subsystem ID */
-	11,		/* Number of entries */
+	0x10ec0269,	/* Codec Vendor/Device ID: Realtek ALC269 */
+	0x10ec0269,	/* Subsystem ID */
+	12,		/* Number of entries */
 
-	AZALIA_SUBVENDOR(0, 0x10ec0000),
+	AZALIA_RESET(0x1),
+
+	AZALIA_SUBVENDOR(0, 0x10ec0269),
 	AZALIA_PIN_CFG(0, 0x12, 0x411111f0), /* NC */
 	AZALIA_PIN_CFG(0, 0x14, 0x411111f0), /* NC */
 	AZALIA_PIN_CFG(0, 0x15, 0x02211010), /* Jack analog out */
 	AZALIA_PIN_CFG(0, 0x17, 0x411111f0), /* NC */
-	AZALIA_PIN_CFG(0, 0x18, 0x02a11020), /* Jack analog mic */
+	AZALIA_PIN_CFG(0, 0x18, 0x02a11120), /* Jack analog mic, no presence detect */
 	AZALIA_PIN_CFG(0, 0x19, 0x411111f0), /* NC */
 	AZALIA_PIN_CFG(0, 0x1a, 0x411111f0), /* NC */
 	AZALIA_PIN_CFG(0, 0x1b, 0x411111f0), /* NC */
-- 
2.30.2

